jans-cedarling/cedarling/src/authz/entities/entity_create.rs

```diff
@@ -0,0 +1,140 @@
+/*
+ * This software is available under the Apache-2.0 license.
+ * See https://www.apache.org/licenses/LICENSE-2.0.txt for full text.
+ *
+ * Copyright (c) 2024, Gluu, Inc.
+ */
+
+use std::{
+    collections::{HashMap, HashSet},
+    str::FromStr,
+};
+
+use crate::models::token_data::{GetTokenClaimValue, TokenClaim};
+use cedar_policy::{EntityId, EntityTypeName, EntityUid, RestrictedExpression};
+
+/// Meta information about an entity type.
+/// Is used to store in `static` variable.
+pub struct EntityMetaInfo<'a> {
+    entity_type: &'a str,
+    entity_id_data_key: &'a str,
+    meta_attributes: Vec<EntityAttributeMeta<'a>>,
+}
+
+impl<'a> EntityMetaInfo<'a> {
+    pub fn new(
+        entity_type: &'a str,
+        entity_id_data_key: &'a str,
+        meta_attributes: Vec<EntityAttributeMeta<'a>>,
+    ) -> Self {
+        Self {
+            entity_type,
+            entity_id_data_key,
+            meta_attributes,
+        }
+    }
+
+    /// Create entity from token data.
+    //
+    // we also can create entity using the ['create_entity'] function.
+    pub fn create_entity(
+        &self,
+        data: &TokenClaim,
+        parents: HashSet<EntityUid>,
+    ) -> Result<cedar_policy::Entity, CedarPolicyCreateTypeError> {
+        create_entity(
+            &self.entity_type,
+            &self.entity_id_data_key,
+            self.meta_attributes.as_slice(),
+            data,
+            parents,
+        )
+    }
+}
+
+/// Meta information about an attribute for cedar policy.
+pub struct EntityAttributeMeta<'a> {
```

Comment:
same thing for this too. `EntityAttributeMetadata`

Reply:
thanks, updated